unit-tests/src/test/java/org/eclipse/collections/impl/factory/ListsTest.java

```diff
@@ -135,6 +138,74 @@ public void emptyList()
     }
 
     @Test
+    public void ofInitialCapacity()
+    {
+        MutableList<String> list1 = Lists.mutable.ofInitialCapacity(0);
+        this.assertPresizedListEquals(0, (FastList<String>) list1);
+
+        MutableList<String> list2 = Lists.mutable.ofInitialCapacity(5);
+        this.assertPresizedListEquals(5, (FastList<String>) list2);
+
+        MutableList<String> list3 = Lists.mutable.ofInitialCapacity(20);
+        this.assertPresizedListEquals(20, (FastList<String>) list3);
+
+        MutableList<String> list4 = Lists.mutable.ofInitialCapacity(60);
+        this.assertPresizedListEquals(60, (FastList<String>) list4);
+
+        MutableList<String> list5 = Lists.mutable.ofInitialCapacity(64);
+        this.assertPresizedListEquals(64, (FastList<String>) list5);
+
+        MutableList<String> list6 = Lists.mutable.ofInitialCapacity(65);
+        this.assertPresizedListEquals(65, (FastList<String>) list6);
+
+        Verify.assertThrows(IllegalArgumentException.class, () -> Lists.mutable.ofInitialCapacity(-12));
+    }
+
+    @Test
+    public void withInitialCapacity()
+    {
+        MutableList<String> list1 = Lists.mutable.withInitialCapacity(0);
+        this.assertPresizedListEquals(0, (FastList<String>) list1);
+
+        MutableList<String> list2 = Lists.mutable.withInitialCapacity(14);
+        this.assertPresizedListEquals(14, (FastList<String>) list2);
+
+        MutableList<String> list3 = Lists.mutable.withInitialCapacity(17);
+        this.assertPresizedListEquals(17, (FastList<String>) list3);
+
+        MutableList<String> list4 = Lists.mutable.withInitialCapacity(25);
+        this.assertPresizedListEquals(25, (FastList<String>) list4);
+
+        MutableList<String> list5 = Lists.mutable.withInitialCapacity(32);
+        this.assertPresizedListEquals(32, (FastList<String>) list5);
+
+        Verify.assertThrows(IllegalArgumentException.class, () -> Lists.mutable.withInitialCapacity(-6));
+    }
+
+    private void assertPresizedListEquals(int initialCapacity, FastList<String> list)
+    {
+        try
+        {
+            Field itemsField = FastList.class.getDeclaredField("items");
```

Comment:
These will work with Java-9 currently but will show a warning. We need to add this check:
`Assume.assumeTrue(System.getProperty("java.version").startsWith("1.8."));`